queries/idris/highlights.scm

```diff
@@ -0,0 +1,208 @@
+; ------------------------------------------------------------------------------
+; Literals and comments
+(integer) @number
+
+(literal
+  (number)) @number.float
+
+(char) @character
+
+[
+  (string)
+  (pat_string)
+  (triple_quote_string)
+] @string
+
+(comment) @comment
+
+; ------------------------------------------------------------------------------
+; Punctuation
+[
+  "("
+  ")"
+  "{"
+  "@{"
+  "}"
+  "["
+  "[<"
+  "]"
+] @punctuation.bracket
+
+[
+  (comma)
+  (colon)
+  (pat_op)
+  (tuple_operator)
+] @punctuation.delimiter
+
+(pat_name
+  (loname) @variable.parameter)
+
+; ------------------------------------------------------------------------------
+; Types
+(signature
+  (loname) @type)
+
+(signature
+  (caname) @constructor)
+
+(caname) @constructor
+
+; ------------------------------------------------------------------------------
+; Keywords, operators, imports
+[
+  "if"
+  "then"
+  "else"
+  "case"
+  "of"
+] @keyword.conditional
+
+(module) @module
+
+[
+  "import"
+  "module"
+  "namespace"
+  "parameters"
+] @keyword.import
+
+[
+  (operator)
```

Comment:
Operators that are English words should be `@keyword.operator` (same in `ipkg`).